### PR TITLE
Gt4py is used as a submodule

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR updates pace to use gt4py pinned as a submodule, rather than using GT4PY_VERSION.txt. After this PR, the gt4py caches will use the gt4py sha rather than tag in their GT4PY_VERSION.txt files.

### CODE REVIEW

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

File: docker/Makefile [hunk 1]
```diff
@@ -3,35 +3,24 @@ SHELL := /bin/bash
...
-.PHONY: dev fv3gfs_image build build_deps push_deps pull_deps get_gt4py
-
-get_gt4py:
-	if [ ! -f ${GT4PY_DIR}/pyproject.toml ]  || \
-	[ "$(shell cd ${GT4PY_DIR} && git describe --tags )" != "${GT4PY_VERSION}"  ]; then \
-                rm -rf ${GT4PY_DIR} && \
-	  	git clone https://github.com/ai2cm/gt4py.git ${GT4PY_DIR} && \
-	  	cd ${GT4PY_DIR} && git checkout ${GT4PY_VERSION};\
-	fi
+.PHONY: dev fv3gfs_image build build_deps push_deps pull_deps
```
Comment (eddie-c-davis) [inline docker/Makefile:20]: So much cleaner!

File: .gitmodules [hunk 2]
```diff
@@ -4,6 +4,9 @@
 [submodule "external/fv3gfs-wrapper"]
 	path = external/fv3gfs-wrapper
 	url = git@github.com:ai2cm/fv3gfs-wrapper.git
+[submodule "external/gt4py"]
```
Comment (eddie-c-davis) [inline .gitmodules:7]: Is this where specific tags would be indicated (e.g., the sha for `v36`)?

File: external/daint_venv/install.sh [hunk 3]
```diff
@@ -37,17 +37,6 @@ if [ $save_wheel ]; then
 fi
 python3 -m pip install --find-links=$wheeldir cupy Cython clang-format
 
-
-# installation of gt4py
```
Comment (eddie-c-davis) [inline external/daint_venv/install.sh:41]: Nice to see this going away.

**Comment by mcgibbon** [comment]
Launched performance test as https://jenkins.ginko.ch/job/pace-fv3core-performance-test/65/

File: fv3core/external/gt4py [hunk 4]
```diff
@@ -0,0 +1 @@
+../../external/gt4py
```
Comment (eddie-c-davis) [inline fv3core/external/gt4py:1]: Is this one of the symlinks we have discussed?

File: pace-util/constraints.txt [hunk 5]
```diff
@@ -1,100 +1,338 @@
 #
-# This file is autogenerated by pip-compile
+# This file is autogenerated by pip-compile with python 3.8
```
Comment (eddie-c-davis) [inline pace-util/constraints.txt:2]: Much easier on the eyes...

**Review by eddie-c-davis — APPROVED** [review]
This is a heroic effort  that simplifies the repository considerably. Have a few questions, but LGTM!

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by rheacangeo** [comment]
What is the plan if we want to run an older commit on pace, but the gt4py-develop commit-hash no longer exists? The previous process involved an automatic tag that would preserve it, will that be a manual process now?

**Comment by mcgibbon** [comment]
> What is the plan if we want to run an older commit on pace, but the gt4py-develop commit-hash no longer exists? The previous process involved an automatic tag that would preserve it, will that be a manual process now?

The plan is to still make commit tags to preserve commits.

**Comment by rheacangeo** [comment]
But it will no longer be automatic, this is a little risky

File: fv3core/external/gt4py (hunk 4 above)
Comment (mcgibbon) [reply fv3core/external/gt4py:1]: I'm not positive I know what you mean, but i think yes?

File: .gitmodules (hunk 2 above)
Comment (mcgibbon) [reply .gitmodules:7]: You'd indicate a tag by checking out that commit in the submodule, and adding it to the repo with `git add external/gt4py` or similar. Checking it out by tag or commit makes no difference.

**Comment by mcgibbon** [comment]
Let's continue the discussion at https://github.com/ai2cm/pace/pull/64.

**Pull request merged**